File: BusinessLogic/Facturacion/Mapping/Cat_Marca.cs

using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;
using APPCORE;
namespace DataBaseModel {
    public class Cat_Marca : EntityClass
    {
        [PrimaryKey(Identity = true)]
        public int? Id_Marca { get; set; }
        public string? Nombre { get; set; }
        public string? Descripcion { get; set; }
        public string? Estado { get; set; }
        //[OneToMany(TableName = "Cat_Producto", KeyColumn = "Id_Marca", ForeignKeyColumn = "Id_Marca")]
        public List<Cat_Producto>? Cat_Producto { get; set; }
    }
}

File: UI/wwwroot/Facturacion/FrontModel/Cat_Marca.js
//@ts-check
import { EntityClass } from "../../WDevCore/WModules/EntityClass.js";
import { Cat_Producto }  from './Cat_Producto.js'
class Cat_Marca extends EntityClass {
   constructor(props) {
       super(props, 'EntityFacturacion');
       Object.assign(this, props);
   }
   /**@type {Number}*/ Id_Marca;
   /**@type {String}*/ Nombre;
   /**@type {String}*/ Descripcion;
   /**@type {String}*/ Estado;
   /**@type {Array<Cat_Producto>} OneToMany*/ Cat_Producto;
}
export { Cat_Marca }
